server/core/src/main/java/org/infinispan/server/core/admin/embeddedserver/CacheCreateTask.java

```diff
@@ -0,0 +1,55 @@
+package org.infinispan.server.core.admin.embeddedserver;
+
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.server.core.admin.AdminServerTask;
+
+/**
+ * Admin operation to create a cache
+ * Parameters:
+ * <ul>
+ *    <li><strong>name</strong> the name of the cache to remove</li>
+ *    <li><strong>flags</strong> </li>
+ * </ul>
+ *
+ * @author Tristan Tarrant
+ * @since 9.1
+ */
+public class CacheCreateTask extends AdminServerTask<Void> {
+
+   @Override
+   public String getTaskContextName() {
+      return "cache";
+   }
+
+   @Override
+   public String getTaskOperationName() {
+      return "create";
+   }
+
+   @Override
+   public Void call() throws Exception {
+      if (isPersistent())
+         throw new UnsupportedOperationException();
+
+      String name = requireParameter("name");
+      String template = getParameter("template");
+      cacheManager.executor().submitConsumer(localManager -> {
```

Comment:
Ideally the tasks should really return a CompletableFuture so they can be handled completely async. Until then, we sync to ensure that the operation has been applied and we return a result (null) :)